test/abstract_syntax_tree.coffee

```diff
@@ -343,6 +343,91 @@ test "AST as expected for CSXTag node", ->
         postfix: yes
       ]
 
+  testExpression '<div>abc</div>',
+    type: 'JSXElement'
+    openingElement:
+      type: 'JSXOpeningElement'
+      name:
+        type: 'JSXIdentifier'
+        name: 'div'
+      attributes: []
+      selfClosing: no
+    closingElement:
+      type: 'JSXClosingElement'
+      name:
+        type: 'JSXIdentifier'
+        name: 'div'
+    children: [
+      type: 'JSXText'
+      extra:
+        raw: 'abc'
+      value: 'abc'
+    ]
+
+  testExpression '''
+    <a>
+      {b}
+      <c />
+    </a>
+  ''',
+    type: 'JSXElement'
+    openingElement:
+      type: 'JSXOpeningElement'
+      name:
+        type: 'JSXIdentifier'
+        name: 'a'
+      attributes: []
+      selfClosing: no
+    closingElement:
+      type: 'JSXClosingElement'
+      name:
+        type: 'JSXIdentifier'
+        name: 'a'
+    children: [
+      type: 'JSXText'
+      extra:
+        raw: '\n  '
+      value: '\n  '
+    ,
+      type: 'JSXExpressionContainer'
+      expression: ID 'b'
+    ,
+      type: 'JSXText'
+      extra:
+        raw: '\n  '
+      value: '\n  '
+    ,
+      type: 'JSXElement'
+      openingElement:
+        type: 'JSXOpeningElement'
+        name:
+          type: 'JSXIdentifier'
+          name: 'c'
+        selfClosing: true
+      closingElement: null
+      children: []
+    ,
+      type: 'JSXText'
+      extra:
+        raw: '\n'
+      value: '\n'
+    ]
+
+  testExpression '<>abc{}</>',
+    type: 'JSXFragment'
+    openingFragment:
+      type: 'JSXOpeningFragment'
+    closingFragment:
+      type: 'JSXClosingFragment'
+    children: [
+      type: 'JSXText'
+      extra:
+        raw: 'abc'
+      value: 'abc'
+    ,
+      type: 'JSXEmptyExpression'
+    ]
```

Comment:
I assume all these match Babel? Since there’s nothing custom about JSX in CoffeeScript.

Reply:
Yup exactly, they should all match Babel's JSX AST. We've discussed converting all `CSX*` names to `JSX*` as part of this, which would simplify the JSX AST code a bit in that they'd all have node class names matching the AST type and so wouldn't need custom `astType()` methods

This PR is the last chunk of JSX AST coverage, so should I go ahead and make a PR for `CSX` -> `JSX` now? Or better to wait until after we've gotten through the rest of the AST stuff?

Reply:
Do you mind following up this PR with another one right after it that just does the rename? That would be good to keep in its own commit for posterity, and for being able to see in the history what changed due to that versus the AST work.